LIFECYCLE.md

```diff
@@ -0,0 +1,55 @@
+| `Devfile Team`          | The team that is responsible for onboarding vendor devfiles for both the community and product registries.                              |
+| `Stack/Sample Provider` | The person or team that is responsible for developing the devfile stack and/or that is available for sharing on our devfile registries. |
+| `Tooling Clients`       | These are the tools that support building devfile-based applications                                                                    |
+| `End Users`             | Application developers that consume devfiles for the purpose of building their own custom applications                                  |
+
+The following sections describe the steps that can happen in the lifecycle of a devfile.
+
+## [Onboarding](CONTRIBUTING.md)
+When a `Stack Provider` is ready to share their devfile to the public/product registry, they must follow the steps in the [contributing guide](CONTRIBUTING.md) to ensure there’s basic information that identifies the version, owner, description, etc of the stack and ensure the stack meets minimal testing requirements against the supported clients.
+
+| Role             | Responsibilities                                                                                                                                                                                                                                                                                                           |
+|:-----------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Stack Provider` | <ul>Follow the [contributing guide](CONTRIBUTING.md) and ensure stack is tested against the supported clients </ul>                                                                                                                                                                                                        |
+| `Devfile Team`   | <ul><li>Review the PR</li><li>Work with `Tooling Clients` to ensure there is sufficient testing</li><li>Ensure the stack has the right contact information and any 2.2 and above versions contain the support link</li><li>Responsible for giving the stack owner repo permissions to modify the stacks they own</li></ul> |
+
+
+## Maintenance
+During the course of its lifecycle, a stack or sample may need to be updated by the `Stack Provider`.  Types of updates can include:
+
+* Changing the devfile content itself where components, commands, resources are modified
+* Using a more secure container image(s)
+* Implementing a new devfile spec version of the existing devfile e.g. two devfiles exist for the same runtime supporting both inner and outer loop scenarios.
```

Comment:
I'm not sure the example given clearly match this scenario and sounds more like a case of having multiple devfiles under the same workspace. We could include this as an update scenario of its own. In addition, I think an example for spec version would be:
> e.g. multiple versions of devfiles exist for the same runtime supporting different versions of devtools

An additional scenario for the inner and outerloop scenarios could go like:
> Implementing new additional devfile to the current stack or sample e.g. two devfiles, `devfile.yaml` and `prod.devfile.yaml`, exist for the same runtime supporting both inner and outer loop scenarios.

Reply:
Yup +1 from me. I think having two different example might make things clearer. I've added your suggestions to the doc. I'm resolving the comment, if you think it needs more refinement we can always re-open!